cli/dotfiles.go

```diff
@@ -140,6 +146,23 @@ func (r *RootCmd) dotfiles() *clibase.Cmd {
 				_, _ = fmt.Fprintln(inv.Stdout, cliui.DefaultStyles.Error.Render("Failed to update repo, continuing..."))
 			}
 
+			if dotfilesExists && gitbranch != "" {
+				// If the repo exists and the git-branch is specified, we need to check out the branch. We do this after
+				// git pull to make sure the branch was pulled down locally. If we do this before the pull, we could be
+				// trying to checkout a branch that does not yet exist locally and get a git error.
```

Comment:
Git pull will only forward the current branch. The requested branch may be available locally but not forwarded. If we pull main and checkout other, other will still be stale until the next time this runs (judging from this comment, didn't look at the code yet).

Reply:
Correct, but this code does a `git checkout + git pull` if we are on the wrong branch.

So the first step is either a `git clone` or a `git pull`. 
The second step is a `git checkout + git pull` **if and only if** `git branch --show-current` is not the correct branch.

Tl;dr we run 2 pulls